reporter/symbol_uploader.go

```diff
@@ -119,20 +120,27 @@ func NewDatadogSymbolUploader(cfg *SymbolUploaderConfig) (*DatadogSymbolUploader
 		return nil, fmt.Errorf("failed to create cache: %w", err)
 	}
 
+	compressDebugSections := !cfg.DisableDebugSectionCompression && CheckObjcopyZstdSupport()
+
 	return &DatadogSymbolUploader{
-		symbolEndpoints:      cfg.SymbolEndpoints,
-		intakeURLs:           intakeURLs,
-		version:              cfg.Version,
-		dryRun:               cfg.DryRun,
-		uploadDynamicSymbols: cfg.UploadDynamicSymbols,
-		uploadGoPCLnTab:      cfg.UploadGoPCLnTab,
-		client:               &http.Client{Timeout: uploadTimeout},
-		uploadCache:          uploadCache,
-		symbolQueriers:       symbolQueriers,
-		symbolQueryInterval:  cfg.SymbolQueryInterval,
+		symbolEndpoints:       cfg.SymbolEndpoints,
+		intakeURLs:            intakeURLs,
+		version:               cfg.Version,
+		dryRun:                cfg.DryRun,
+		uploadDynamicSymbols:  cfg.UploadDynamicSymbols,
+		uploadGoPCLnTab:       cfg.UploadGoPCLnTab,
+		client:                &http.Client{Timeout: uploadTimeout},
+		uploadCache:           uploadCache,
+		symbolQueriers:        symbolQueriers,
+		symbolQueryInterval:   cfg.SymbolQueryInterval,
+		compressDebugSections: compressDebugSections,
 	}, nil
 }
 
+func CheckObjcopyZstdSupport() bool {
```

Comment:
```suggestion
func checkObjcopyZstdSupport() bool {
```
All other function names start with a lowercase letter.

Reply:
In Go, function/types that starts with a lowercase letter are private to the package and not exported, but here I need to access `CheckObjcopyZstdSupport` in extract_symbols main package.

Reply:
Oh thanks! I did not know that. Makes sense then!